HW04.Rmd

```diff
@@ -1,126 +1,343 @@
+
+# generate bar graph showing the number of gun deaths associated with each type of intent from highest to lowest value
+ggplot(gun_deaths_intent, aes(x = intent, y = number_of_deaths, color = intent, fill = intent)) +
+    geom_col() +
+    labs(title = "Number of Gun Deaths based on Intent", subtitle = "Based on Data from 2012 - 2014", x = "Intent", y = "Number of Deaths") + #adjust labels
+    theme_classic() + # set theme
+    theme(legend.position = "none") # remove legend
+
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+
+# generate new dataframe looking at the age of gun death victims by sex
+gun_deaths_sex <- gun_deaths %>%
+    select(sex, age) %>%
+    group_by(sex) 
+
+# change data type of age column to integer
+gun_deaths_sex$age <- as.integer(gun_deaths_sex$age)
+
+# generate boxplot 
+ggplot(gun_deaths_sex) +
+    geom_boxplot(aes(sex, age, fill = sex)) +
+    labs(title = "Age of Gun Death Victims by Sex", x = "Sex", y = "Age") + # adjust labels
+    theme_bw() + # set theme
+    theme(legend.position = "none") # remove legend
+    
+
+# find the average age of female gun death victims
+gun_deaths_average_female <- gun_deaths_sex %>%
+    filter(sex == "F") %>%
+    summarize(average_age = mean(age, na.rm = TRUE)) # na.rm allows to 'remove' NA values
+
+head(gun_deaths_average_female)
+
+```
+
+
+**The average female gun death victim age was ~43.7 years.**
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+
+# generate new dataframe looking at how many white males with at least a high school education were killed by guns in 2012
+gun_deaths_education <- gun_deaths %>%
+    select(sex, education, year, race) %>%
+    filter(sex == "M", year == 2012, race == "White", education %in% c('HS/GED', 'Some college', 'BA+')) %>% # I believe that this should include all education levels with at least a high school education
+    count()
+
+# print the number of white males killed as kable() output
+kable(gun_deaths_education, format = "markdown", align = 'c', col.names = c("Number of White Males killed"))
+
+```
+
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+
+# generate new dataframe to look at the season of the year with the most gun deaths
+gun_deaths_season <- gun_deaths %>%
+    select(month) 
+
+# convert the 'month' variable to a factor
+gun_deaths_season$month <- as.factor(gun_deaths_season$month)
+
+# convert from months to season
+gun_deaths_season%>%
+    mutate(season = fct_collapse(.f = month,
+      Spring = c("4", "5", "6"),
+      Summer = c("7", "8", "9"),
+      Autumn = c("10", "11", "12"),
+      Winter = c("1", "2", "3"))) %>%
```

Comment:
fct_collapse() is such a nice way of converting months to seasons. I used nested ifelse to accomplish this and it works as well, but your way is much cleaner, since it makes use of an dedicated function.